internal/allowed/topic/create_test.go

```diff
@@ -53,119 +54,131 @@ func assertCreateIsAllowed(t *testing.T, params *allowed.IsAllowedParams) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{}
-	dp.AddUserWithAdminGroupToMeeting(1, 1)
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateFailWithError(t, params)
-}
-
-func TestCreateUserNotInMeeting(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	dp.AddUser(1)
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsNotAllowed(t, params)
-}
-
-func TestCreateAdminUser(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	dp.AddUserWithAdminGroupToMeeting(1, 1)
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsAllowed(t, params)
-}
-
-func TestCreateUser(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	dp.AddUserToMeeting(1, 1)
-	dp.AddPermissionToGroup(1, "agenda.can_manage")
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsAllowed(t, params)
-}
-
-func TestCreateUserNoPermissions(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	dp.AddUserToMeeting(1, 1)
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsNotAllowed(t, params)
-}
-
-func TestCreateInvaldFields(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	dp.AddUserWithSuperadminRole(1)
-	data := definitions.FqfieldData{
-		"not_allowed": "some value",
-	}
-	params := &allowed.IsAllowedParams{UserID: 1, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateFailWithError(t, params)
-}
-
-func TestCreateDisabledAnonymous(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	params := &allowed.IsAllowedParams{UserID: 0, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsNotAllowed(t, params)
-}
-
-func TestCreateEnabledAnonymous(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	dp.EnableAnonymous()
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	params := &allowed.IsAllowedParams{UserID: 0, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsNotAllowed(t, params)
-}
-
-func TestCreateEnabledAnonymousWithPermissions(t *testing.T) {
-	dp := tests.NewTestDataProvider()
-	dp.EnableAnonymous()
-	dp.AddPermissionToGroup(1, "agenda.can_manage")
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	params := &allowed.IsAllowedParams{UserID: 0, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsAllowed(t, params)
+func TestCreate(t *testing.T) {
+	t.Run("UnknownUser", func(t *testing.T) {
+		dp := tests.NewTestDataProvider(context.TODO())
```

Comment:
From a technical view, `context.TODO()` and `context.Background()` do exactly the same. The Name `TODO` should only be used, when there is something, the developer should do.

I don't know, if you want to do something else here in the near future. In this case, the context `context.TODO()` is correct. But if you want this code to stay this way, then `context.Background()` would be the right choice.